### ISSUE TITLE
help: auto-generate $values, $description for errors

### ISSUE THREAD

**Issue by catb0t** [factor/factor#1928]
```factor
HELP: ordinary-word-missing-section
{ $values { "missing-section" string } { "word-name" string } }
{ $description "Throws an " { $link ordinary-word-missing-section } " error." }
{ $error-description "Thrown when an ordinary word's documentation is missing a required section." } ;
```
```factor
HELP: no-method
{ $values { "object" object } { "generic" "a generic word" } }
{ $description "Throws a " { $link no-method } " error." }
{ $error-description "Thrown by the " { $snippet "generic" } " word to indicate it does not have a method for the class of " { $snippet "object" } "." } ;
```
The top two elements, `$values` and `$description`, should be auto-generated based on the word's definition by the help system, like the help system can autogenerate `$values` for ordinary words. (surprisingly, they aren't autogenerated)

The `$description` element will always say "Throws an **error** error.", but if it needs to be different then it should be explicit.

That way, all we as docs writers have to do is write an `$error-description` which is actually error-specific.

**Comment by mrjbq7** [comment]
Maybe you could do something like this:

```diff
diff --git a/basis/tools/scaffold/scaffold.factor b/basis/tools/scaffold/scaffold.factor
index 762dcd6d25..900a4fe26f 100644
--- a/basis/tools/scaffold/scaffold.factor
+++ b/basis/tools/scaffold/scaffold.factor
@@ -1,11 +1,11 @@
 ! Copyright (C) 2008 Doug Coleman.
 ! See http://factorcode.org/license.txt for BSD license.
 USING: accessors alien arrays assocs byte-arrays calendar
-classes combinators combinators.short-circuit fry hashtables
-help.markup interpolate io io.directories io.encodings.utf8
-io.files io.pathnames io.streams.string kernel math math.parser
-namespaces prettyprint quotations sequences sets sorting
-splitting strings system timers unicode urls vocabs
+classes classes.error combinators combinators.short-circuit fry
+hashtables help.markup interpolate io io.directories
+io.encodings.utf8 io.files io.pathnames io.streams.string kernel
+math math.parser namespaces prettyprint quotations sequences
+sets sorting splitting strings system timers unicode urls vocabs
 vocabs.loader vocabs.metadata words words.symbol ;
 IN: tools.scaffold
 
@@ -176,6 +176,12 @@ M: object add-using ( object -- )
         ] if
     ] when* ;
 
+: error-description. ( word -- )
+    [ $values. ] [
+        "{ $description \"Throws an \" { $link " write
+        name>> write " } \" error.\" }" print
+    ] bi "{ $error-description \"\" } ;" print ;
+
 : class-description. ( word -- )
     drop "{ $class-description \"\" } ;" print ;
 
@@ -187,6 +193,7 @@ M: object add-using ( object -- )
 
 : docs-body. ( word/symbol -- )
     {
+        { [ dup error-class? ] [ error-description. ] }
         { [ dup class? ] [ class-description. ] }
         { [ dup symbol? ] [ symbol-description. ] }
         [ [ $values. ] [ $description. ] bi ]
```

**Comment by mrjbq7** [comment]
Plus/minus proper "a/an" grammar.

**Comment by mrjbq7** [comment]
Fixed in 1685c4537cccbf76fc68abf1125ff3be57d11b25.
